crates/bifrost/benches/append_throughput.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::ops::Range;
+
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+use futures::stream::{FuturesOrdered, FuturesUnordered};
+use futures::StreamExt;
+use restate_bifrost::{Bifrost, BifrostService};
+use restate_core::metadata;
+use restate_rocksdb::{DbName, Owner, RocksDbManager};
+use restate_types::config::{
+    BifrostOptionsBuilder, CommonOptionsBuilder, ConfigurationBuilder, LocalLogletOptionsBuilder,
+};
+use restate_types::logs::{LogId, Payload};
+use tracing::info;
+use tracing_subscriber::EnvFilter;
+mod util;
+
+async fn append_records_multi_log(bifrost: Bifrost, log_id_range: Range<u64>, count_per_log: u64) {
+    // for every log there is N streams. total count_per_log / streams_per_log gives the number of
+    // writes per stream per log.
+    let mut appends = FuturesUnordered::new();
+    for log_id in log_id_range {
+        for _ in 0..count_per_log {
+            let mut bifrost = bifrost.clone();
+            appends.push(async move {
+                let _ = bifrost
+                    .append(LogId::from(log_id), Payload::default())
+                    .await
+                    .unwrap();
+            })
+        }
+    }
+    while appends.next().await.is_some() {}
+}
+
+async fn append_records_concurrent_single_log(bifrost: Bifrost, log_id: LogId, count_per_log: u64) {
+    // for every log there is N streams. total count_per_log / streams_per_log gives the number of
+    // writes per stream per log.
+    let mut appends = FuturesOrdered::new();
+    for _ in 0..count_per_log {
+        let mut bifrost = bifrost.clone();
+        appends.push_back(async move { bifrost.append(log_id, Payload::default()).await.unwrap() })
+    }
+    while appends.next().await.is_some() {}
+}
+
+async fn append_seq(mut bifrost: Bifrost, log_id: LogId, count: u64) {
+    for _ in 1..=count {
+        let _ = bifrost
+            .append(log_id, Payload::default())
+            .await
+            .expect("bifrost accept record");
+    }
+}
+
+fn write_throughput_local_loglet(c: &mut Criterion) {
+    let test_runner_rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .expect("current thread runtime must build");
+
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .init();
+
+    let tmp_base = tempfile::tempdir().unwrap();
+
+    let provider = restate_types::logs::metadata::ProviderKind::Local;
+    let total_num_logs = 64;
+    let num_logs_per_run = 10;
```

Comment:
Possible follow-up: What I found helpful when playing around with different configurations is to make these values configurable via env variables or so. That way one does not have to rebuild the benchmark every time one wants to test a different configuration.

Reply:
Absolutely, good idea, might do that on a follow up.